lib/srv/app/azure/handler.go

```diff
@@ -293,6 +293,10 @@ func (s *handler) getToken(ctx context.Context, managedIdentity string, scope st
 	var tokenResult *azcore.AccessToken
 	var errorResult error
 
+	// call Clock.After() before FnCacheGet gets called in a different go-routine.
+	// this ensures there is no race condition in the timeout tests
```

Comment:
How does this change anything? `After` doesn't block, and neither does the goroutine.

Reply:
`After` doesn't block, but the channel it returns does.

In the old code we have `clock.Advance(getTokenTimeout)` and `clock.After(getTokenTimeout)` racing each other on two separate go-routines. If `clock.After(getTokenTimeout)` wins, then subsequent call to `clock.Advance(getTokenTimeout)` will unblock the returned channel. Otherwise, the channel will unblock only at `getTokenTimeout+getTokenTimeout`, which doesn't happen.

You can force the `clock.Advance` to win the race by adding `time.Sleep(time.Second)` before `select` and after `go func()`, i.e. around line 306. Given that change, the old code would always deadlock while the new one does not: we ensure there is no race by calling `After` before the go-routine is spawned.

Reply:
```suggestion
	// call Clock.After() before FnCacheGet gets called in a different go-routine.
	// this ensures there is no race condition in the timeout tests, as
	// getAccessToken() ends up calling Clock.Advance() there
```